src/unix/thread.c

```diff
@@ -50,6 +62,16 @@ sysreturn clone(unsigned long flags, void *child_stack, int *ptid, int *ctid, un
         return set_syscall_error(current, ENOSYS);
     }
 
+    /* no stack size given, just validate the top word */
+    if (!validate_user_memory(child_stack, sizeof(u64), true))
+        return set_syscall_error(current, EFAULT);
```

Comment:
As far as I can see, the kernel will never access this stack, so I think it shouldn't bother validating it.

Reply:
I'm not sure if there's any reason for the kernel to access the user stack. I can't think of one either...but I'd still prefer a little paranoia to mask any unintended affects from a bad address getting loaded into %rsp and then a return to the frame. I can't see how the validate could hurt.